mm2src/coins/rpc_command/tendermint/staking.rs

```diff
@@ -0,0 +1,149 @@
+use common::{HttpStatusCode, PagingOptions, StatusCode};
+use cosmrs::staking::{Commission, Description, Validator};
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::MmError;
+
+use crate::{lp_coinfind_or_err, tendermint::TendermintCoinRpcError, MmCoinEnum};
+
+/// Represents current status of the validator.
+#[derive(Default, Deserialize)]
+pub(crate) enum ValidatorStatus {
+    All,
+    /// Validator is in the active set and participates in consensus.
+    #[default]
+    Bonded,
+    /// Validator is not in the active set and does not participate in consensus.
+    /// Accordingly, they do not receive rewards and cannot be slashed.
+    /// It is possible to delegate tokens to a validator in this state.
+    Unbonded,
+}
+
+impl ToString for ValidatorStatus {
+    fn to_string(&self) -> String {
+        match self {
+            ValidatorStatus::All => String::default(),
```

Comment:
i think the empty string here is intentional as `QueryValidatorsRequest` expects that?

leaving a comment about this would be helpful. we might also wanna consider having this an `impl` method (e.g. `status.serialize_for_request()` or something) instead of implementing `ToString`.

Reply:
Left a comment.